tests/trainer/properties/test_auto_gpu_select.py

```diff
@@ -59,3 +60,23 @@ def test_pick_multiple_gpus(nb, expected_gpu_idxs, expected_error):
             pick_multiple_gpus(nb)
     else:
         assert expected_gpu_idxs == pick_multiple_gpus(nb)
+
+
+@RunIf(min_gpus=2)
+@mock.patch("pytorch_lightning.trainer.connectors.accelerator_connector.pick_multiple_gpus", return_value=[1])
+def test_auto_select_gpus_with_devices_flag(_):
+
+    trainer = Trainer(auto_select_gpus=True, accelerator="gpu", devices=1)
+    assert trainer.num_devices == 1
+    assert trainer.device_ids == [1]
+
+
+@RunIf(min_gpus=2)
+@mock.patch("pytorch_lightning.trainer.connectors.accelerator_connector.pick_multiple_gpus", return_value=[1])
+def test_auto_select_gpus_with_gpus_flag(_):
+
```

Comment:
It woudl make sense to me to merge both tests. It will also be a tiny bit faster

```suggestion
```

Reply:
Done!